test/integration/integration_test.go

```diff
@@ -548,6 +551,404 @@ func (i *integrationTest) exerciseIncrementalBuild(tag, imageName string, remove
 	}
 }
 
+func TestDockerfileBuild(t *testing.T) {
+	tempdir, err := ioutil.TempDir("", "s2i-dockerfiletest-dir")
+	if err != nil {
+		t.Errorf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tempdir)
+
+	config := &api.Config{
+		BuilderImage:    "docker.io/centos/nodejs-8-centos7",
+		AssembleUser:    "",
+		ImageWorkDir:    "",
+		ImageScriptsDir: "/usr/libexec/s2i",
+		Source:          git.MustParse("https://github.com/sclorg/nodejs-ex"),
+		ScriptsURL:      "",
+		Injections:      api.VolumeList{},
+		Destination:     "",
+
+		Environment: api.EnvironmentList{},
+		Labels:      map[string]string{},
+
+		AsDockerfile: tempdir + string(os.PathSeparator) + "Dockerfile",
```

Comment:
One of these tests should have `AsDockerfile: tempdir` as the input (`Build` should mutate the value to `tmpdir/Dockerfile`)

Reply:
will add